src/core/CoreSession.jl

```diff
@@ -5,61 +5,146 @@ const PULSE_INTERVAL_MILLIS = 5000
 mutable struct  CoreSession <: AbstractCoreSession
     client::CoreClient
     database::CoreDatabase
-    sessionID::Array{UInt8,1}
-    transactions::Array{Union{Nothing,<:AbstractCoreTransaction},1}
+    sessionID::Bytes
+    transactions::Dict{UUID,T} where {T<:Union{Nothing,<:AbstractCoreTransaction}}
```

Comment:
You can probably write:
```suggestion
    transactions::Dict{UUID,T} where {T<:Optional{AbstractCoreTransaction}}
```

Also, I think it's more common to put the `where` clause in `CoreSession` itself, so you you have `CoreSession{T}`.